### ISSUE TITLE
[Bug] Can't change capitalization in folder name without changing the name.

### ISSUE THREAD

**Issue by HoloWise** [xivdev/Penumbra#157]
Current implementation of Folder Renaming doesn't allow for changes in capitalization ("FOLDER" -> "Folder") since Exists check is not case sensitive (and so returns true and method exits without renaming the folder)

**Comment by Ottermandias** [comment]
Yes. Because windows does that.

**Comment by HoloWise** [comment]
@Ottermandias 

I don't see a reason why that'd be closed? Windows doesn't do that, it's completely fine to rename a folder by just changing capitalization.

If you are renaming a folder then there's no way there'll still be a matching folder once a rename is complete so it's a safe operation and changing capitalization without changing letters is a valid use case.

```
                if( newDir.Exists )
                {
                    return false;
                }
```

This should be expanded to cover a case where newDir.Exists == true but new FullName is same as old FullName to allow for capitalization changes.

I can work on that when I have time if it's too trivial at this stage of development.

**Comment by Ottermandias** [comment]
No, it is not.

**Comment by HoloWise** [comment]
Comparison is case insensitive, names in file systems are stored as case sensitive.

if newDir and oldDir are the same (case insensitive) then all you have to do is rename to temporary name and then rename to new name that has changed capitalization, it's basic renaming logic.

But w/e

**Comment by Ottermandias** [comment]
NTFS is case-insensitive on windows as a file system, cf. https://en.wikipedia.org/wiki/NTFS

**Comment by HoloWise** [comment]
Sorry, I worded that poorly. What I meant is that Windows allows for changes in capitalization and it's a simple renaming logic to achieve that effect in code with no issues even when file systems are not case sensitive, if dir exists, and it's the same dir but with changed capitalization just change name to a temporary name and then to desired name with changed capitalization.

Also sorry my previous comments were kinda aggressive since I see it as a simple QoL fix, it was really unnecessary.

**Comment by Ottermandias** [comment]
I'm not sure it's as easy as you think.
The code snippet you found isn't the thing that's actually stopping the rename, 
```cs
private void RenameModFolder( string newName )
        {
            _newName = newName.ReplaceBadXivSymbols();
            if( _newName.Length == 0 )
            {
                PluginLog.Debug( "New Directory name {NewName} was empty after removing invalid symbols.", newName );
                ImGui.CloseCurrentPopup();
            }
            else if( !string.Equals( _newName, Mod!.Data.BasePath.Name, StringComparison.InvariantCultureIgnoreCase ) )
            {
                var           dir    = Mod!.Data.BasePath;
                DirectoryInfo newDir = new(Path.Combine( dir.Parent!.FullName, _newName ));

                if( newDir.Exists )
                {
                    ImGui.OpenPopup( LabelOverWriteDir );
                }
                else if( _modManager.RenameModFolder( Mod.Data, newDir ) )
                {
                    _selector.ReloadCurrentMod();
                    ImGui.CloseCurrentPopup();
                }
            }
        }
```
This is. I could just compare for case-sensitivity here, Then it would go into the overwrite-dialogue if the folder exists. I would have to treat this differently then, depending on if the file system is case sensitive or isn't, I think?
I actually don't know what would happen with my recursive merge function if it's pointed at the same folder.

**Comment by HoloWise** [comment]
Right, that snippet makes more sense. Thanks for pointing me to the right one. 😄 

I would have to read more into the code to think of a proper solution since I don't code in C# often but based on a quick glance maybe you could use StringComparison.InvariantCulture so check is case sensitive, then if newDir.Exists returns True check it it's a valid overwrite or just a capitalization change (that'll avoid using merge) and if it's capitalization just pass it to _modManager.RenameModFolder as usual and implement logic for capitalization change there by using intermediate temporary rename.

**Comment by Ottermandias** [comment]
I have put up a new commit that should handle it, I'm not entirely sure it works fine (especially on a non-windows system).
I also checked that DirectoryInfo.MoveTo does throw if the target is the same folder, regardless of capitalization, so it seems to need the A -> AB -> B workaround.

**Comment by HoloWise** [comment]
Right! Didn't see that commit. I'll have a look tomorrow with a local build since it's a bit late here and I'll let you know if I see anything odd

**Comment by HoloWise** [comment]
@Ottermandias Now that I finally got the repo cloned and can do some proper work and local testing I think we both ended up looking at wrong code. Sorry, maybe I didn't make it clear what I wanted.

What I initially meant is renaming folders in Installed Mods view.

https://user-images.githubusercontent.com/36460595/155389341-7131c23c-b467-4a62-9b79-876e0b66dae1.mp4

This UI is handled by DrawRenameFolderInput in TabInstalledSelector.cs

The issue is in if checks in RenameNoSave in ModFileSystem.cs

`target.Parent!.FindSubFolder( newName, out var preExisting )` check returns True because it uses ModFolderComparer to compare names and this comparer uses StringComparison.InvariantCultureIgnoreCase which stops users from changing capitalization.

Not sure if there's any reason for that comparer to ignore case but changing it to StringComparison.InvariantCulture worked fine for my use case

https://user-images.githubusercontent.com/36460595/155393805-412e0e8e-3d27-4ba6-942c-bb26d46e91f3.mp4

Here's a small state debug for RenameNoSave when comparer is set up to ignore case 

![image](https://user-images.githubusercontent.com/36460595/155394389-affbc750-a819-48b6-a28e-a0a63f05d22a.png)

**Comment by Ottermandias** [comment]
I was thinking you meant the in-penumbra folder system at first, but the 'Exists' check didn't make any sense for that so I was sure you meant filesystem stuff.
I'll take a look if your solution is enough for a clean change. I might've compared case insensitive only due to similarity for actual file system.

**Comment by HoloWise** [comment]
Sounds good 😄 Thanks for looking into it.

and yeah I mentioned Exists because when writing this Issue I first found RenameModFolder in ModManagerEditExtensions and made an assumption that I'm looking at a correct piece of code. Totally my bad, should have included repro video in the first place.

**Comment by Ottermandias** [comment]
I pushed a small commit that should allow you to rename a folder only changing capitalization without changing sort or merge behaviour otherwise (i.e. sort folders are still case insensitive, but you can rename a folder to change its capitalization).
Please check if this is enough and works fine, because I would not like to change behaviour if possible.

**Comment by Ottermandias** [comment]
I pushed a small commit that should allow you to rename a folder only changing capitalization without changing sort or merge behaviour otherwise (i.e. sort folders are still case insensitive, but you can rename a folder to change its capitalization).
Please check if this is enough and works fine, because I would not like to change behaviour if possible.

**Comment by Ottermandias** [comment]
I pushed a small commit that should allow you to rename a folder only changing capitalization without changing sort or merge behaviour otherwise (i.e. sort folders are still case insensitive, but you can rename a folder to change its capitalization).
Please check if this is enough and works fine, because I would not like to change behaviour if possible.

**Comment by HoloWise** [comment]
Tried to do some checks on local build with newest commit.

Renaming works as expected but I encountered an odd issue

When I added an already existing mod it got added to a random folder (Based on what I see Penumbra retains folder paths for deleted mods so maybe that's it and I just did something a while ago that caused it).

Then when I tried to move it around it just started creating copies of itself on the list and I wasn't able to delete any of them, I downloaded a different new mod, installed it, went to list and old mod that was causing issues was gone and I could move and delete newly added mod again. I added mod that was causing issues again and this time it was working properly. Tried to reload Penumbra but that didn't cause issue to repro.
